### PR TITLE
CHOP-14. Verify that jars added to the DCache are jars.

### PR DESCRIPTION
**Pull request by robotoer**
Also cleaned up jar finding code.

@jhlch @lockoff.

### CODE REVIEW

File: src/main/java/com/wibidata/lang/KijiTap.java [hunk 1]
```diff
@@ -55,17 +56,17 @@
  * MapReduce job with the correct input format for reading from a Kiji table,
  * as well as the proper classpath dependencies for MapReduce tasks.
  */
-@SuppressWarnings("rawtypes")
+@SuppressWarnings({ "rawtypes", "serial" })
 public class KijiTap
     extends Tap<JobConf, RecordReader, OutputCollector> {
   private static final Logger LOG = LoggerFactory.getLogger(KijiTap.class);
-  private static final long serialVersionUID = 1L;
```
Comment (hougs) [inline src/main/java/com/wibidata/lang/KijiTap.java]: Why did you remove this?

**Comment by hougs** [comment]
Generally looks good. I'm not sure why you removed the serialized-uuid and suppressed the associated warning. So, explanation, please?

File: src/main/java/com/wibidata/lang/KijiTap.java (hunk 1 above)
Comment (robotoer) [inline src/main/java/com/wibidata/lang/KijiTap.java]: I initially removed it because the serialversionuid is only useful, from what I understand, in the case where you would be serializing classes and storing them for a long time. The UID could then be used to inform java that much later on after the class has been modified that the stored class can't be deserialized into the newer class definition. This, however, is completely bogus in the cascading case where serialization only happens over a network socket, not to any persistent storage.

That being said, we can add it back in for now and use it as a marker to inform other developers that this class is serializable and include a comment that informs them as to why. The UID is also only useful if we manually increment/change it every time that we modify the class. If we don't do that, I think having this UID is only going to cause problems by tricking java's serialization system into thinking that newer versions of the class should be able to deserialize older stored data.

Comment (hougs) [inline src/main/java/com/wibidata/lang/KijiTap.java]: Is it Tap that requires serializability? If so, why would that be required if it is never used by them? I'm worried that if we are required by Tap to implement serializable, then if they don't mean it now, they may in the future.

File: src/main/java/com/wibidata/lang/KijiTap.java [hunk 2]
```diff
@@ -243,4 +198,44 @@ public boolean equals(Object other) {
   public int hashCode() {
     return Objects.hashCode(mTableURI, mScheme, mId);
   }
+
+  private static List<String> findKijiJars(Configuration fsConf) throws IOException {
```
Comment (hougs) [inline src/main/java/com/wibidata/lang/KijiTap.java]: Javadoc?

File: src/main/java/com/wibidata/lang/KijiTap.java [hunk 3]
```diff
@@ -54,18 +55,21 @@
  * A {@link Tap} for reading data from a Kiji table. The tap is responsible for configuring a
  * MapReduce job with the correct input format for reading from a Kiji table,
  * as well as the proper classpath dependencies for MapReduce tasks.
+ *
+ * Note: Warnings about a missing serialVersionUID are ignored here. When KijiTap is serialized,
+ * the result is not persisted anywhere making serialVersionUID not necessary.
```
Comment (hougs) [inline src/main/java/com/wibidata/lang/KijiTap.java]: not neccesary -> unneccesary
